CMakeLists.txt

```diff
@@ -150,20 +150,36 @@ if(NOT CMAKE_BUILD_TYPE)
 endif()
 
 # 3rdparty dependencies
-macro(add_submodule_directory RELPATH)
-  if(NOT EXISTS "${PROJECT_SOURCE_DIR}/${RELPATH}/CMakeLists.txt")
-    message(SEND_ERROR "${RELPATH} is missing, perform a git submodule update")
```

Comment:
I feel like this is still useful to have, can we maybe abstract the call to `FetchContent` so that this check is performed first?

Reply:
While we're at it, the message be improved a bit, imo. In the SYCL CTS we use

```
message(FATAL_ERROR "The git submodule ${RELPATH} is missing.\nTry running `git submodule update --init`.")`.
```

Reply:
Done.

Reply:
Perfect, thanks!